src/installer/pkg/sfx/Microsoft.NETCore.App/ReadyToRun.targets

```diff
@@ -40,22 +41,48 @@
   <!-- Crossgen2 bring-up before SDK 6.0 Preview 2 propagates to the runtime repo. -->
   <!-- https://github.com/dotnet/runtime/issues/48252 -->
 
+  <UsingTask
+    TaskName="SetEnvVar"
+    TaskFactory="RoslynCodeTaskFactory"
+    AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+
+    <ParameterGroup>
+      <Name ParameterType="System.String" Required="true" />
+      <Value ParameterType="System.String" Required="false" />
+    </ParameterGroup>
+
+    <Task>
+      <Using Namespace="System" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+          Environment.SetEnvironmentVariable(Name, Value);
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
   <PropertyGroup>
     <OriginalDotnetRootValue />
-    <DOTNET_ROOT />
   </PropertyGroup>
 
   <Target Name="PatchDotnetRootBeforeRunningCrossgen2" BeforeTargets="_CreateR2RImages">
     <PropertyGroup>
       <OriginalDotnetRootValue>$(DOTNET_ROOT)</OriginalDotnetRootValue>
-      <DOTNET_ROOT>$(RepoRoot)</DOTNET_ROOT>
+      <DotnetSdkRootFile>$([MSBuild]::NormalizePath('$(RepoRoot)', 'artifacts', 'toolset', 'sdk.txt'))</DotnetSdkRootFile>
     </PropertyGroup>
+
+    <ReadLinesFromFile File="$(DotnetSdkRootFile)">
+      <Output TaskParameter="Lines" PropertyName="DotnetRoot"/>
+    </ReadLinesFromFile>
+
+    <Message Importance="High" Text="DotnetSdkRootFile: $(DotnetSdkRootFile)" />
+    <Message Importance="High" Text="DotnetRoot:        $(DotnetRoot)" />
+
+    <SetEnvVar Name="DOTNET_ROOT" Value="$(DotnetRoot)" />
```

Comment:
Is this a temporary change for investigation?

Reply:
Yes, I keep hitting issues in properly reading the DOTNET_ROOT from the sdk.txt file, I'll delete these once the change works.

Reply:
I see that this change has been merged rather than removed. Why could not we use the `DotNetHostPath` metadata that you added to SDK recently?

Reply:
Hi Anton! Can you please clarify your comment? For the "Message" tags, I removed those. For DOTNET_ROOT, that is still necessary because technically the runtime repo still supports SDK preview 1 as the minimum i.e. the one that doesn't yet have my DotNetHostPath support. That will happen as part of the April infra rollout (I think 4/5 is the date) after which I'll be able to merge in my change

https://github.com/dotnet/runtime/pull/49004

cleaning up the hack (modulo the fact that I need to rebase the change after this check-in).

Reply:
My comment related to the `SetEnvVar` line, not the `Message` ones, which GitHub displays as the context. Thank you for the clarification!